atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/TargetedSweepEteTest.java

```diff
@@ -17,28 +17,98 @@
 
+import java.util.Random;
 import java.util.concurrent.TimeUnit;
 
 import org.awaitility.Awaitility;
+import org.junit.After;
+import org.junit.Assert;
 import org.junit.Test;
 
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.todo.ImmutableTodo;
 import com.palantir.atlasdb.todo.Todo;
 import com.palantir.atlasdb.todo.TodoResource;
+import com.palantir.atlasdb.todo.generated.TodoSchemaTableFactory;
 
 public class TargetedSweepEteTest {
     private static final Todo TODO = ImmutableTodo.of("some stuff to do");
+    private static final TodoSchemaTableFactory FACTORY = TodoSchemaTableFactory.of(Namespace.DEFAULT_NAMESPACE);
+    private static final TableReference INDEX_TABLE = FACTORY.getSnapshotsStreamIdxTable(null).getTableRef();
+    private static final TableReference HASH_TABLE = FACTORY.getSnapshotsStreamHashAidxTable(null).getTableRef();
+    private static final TableReference METADATA_TABLE = FACTORY.getSnapshotsStreamMetadataTable(null).getTableRef();
+    private static final TableReference VALUES_TABLE = FACTORY.getSnapshotsStreamValueTable(null).getTableRef();
+
+    private TodoResource todoClient = EteSetup.createClientToSingleNode(TodoResource.class);
+
+    @After
+    public void cleanupStreamTables() {
+        todoClient.truncate();
+    }
 
     @Test
     public void backgroundThoroughSweepDeletesOldVersion() throws InterruptedException {
-        TodoResource todoClient = EteSetup.createClientToSingleNode(TodoResource.class);
-
         long ts = todoClient.addTodoWithIdAndReturnTimestamp(100L, TODO);
         assertThat(todoClient.doesNotExistBeforeTimestamp(100L, ts)).isFalse();
 
         todoClient.addTodoWithIdAndReturnTimestamp(100L, TODO);
         Awaitility.waitAtMost(2, TimeUnit.MINUTES).pollInterval(2, TimeUnit.SECONDS)
                 .until(() -> todoClient.doesNotExistBeforeTimestamp(100L, ts));
     }
+
+    @Test
+    public void targetedSweepSmallStreamsTest() {
+        storeFiveStreams(20);
+        todoClient.runIterationOfTargetedSweep();
+        assertDeleted(4, 4, 4, 4);
+        assertDeletedAndSwept(4, 0, 0, 0);
+
+        storeFiveStreams(20);
+        todoClient.runIterationOfTargetedSweep();
+        assertDeleted(9, 9, 9, 9);
+        assertDeletedAndSwept(9, 4, 4, 4);
+
+        todoClient.runIterationOfTargetedSweep();
+        assertDeleted(9, 9, 9, 9);
+        assertDeletedAndSwept(9, 9, 9, 9);
+    }
+
+    @Test
+    public void targetedSweepLargeStreamsTest() {
+        storeFiveStreams(1500000);
+        todoClient.runIterationOfTargetedSweep();
+        assertDeleted(4, 4, 4, 4 * 4);
+        assertDeletedAndSwept(4, 0, 0, 0);
+
+        todoClient.runIterationOfTargetedSweep();
+        assertDeleted(4, 4, 4, 4 * 4);
+        assertDeletedAndSwept(4, 4, 4, 4 * 4);
+    }
+
+    private void storeFiveStreams(int streamSize) {
```

Comment:
⚠️ looks like you copied this method over from another ETE test. Please find a common place for it.